# 0. Assessment strategy 

### Question 1: State which assessment strategy you have chosen and explain how you have implemented it.

The assessment strategy I have chosen is to acquire a test dataset. To do this, I will use the gui that I created in lab 2 to gather examples of each digit that I can save in a testing set. From there, I will use this set to create predictions with the model I saved in lab 3 and assess the model to see if it accurately predicts the digits.

# 1. Analyze results from production prototype

At least 5 examples for each digit should be available and stored in `lab4-results.csv` and compute:
- accuracy
- confusion matrix 

here.


In [74]:
!pip install -Uqq fastbook
import fastbook
fastbook.setup_book()
from fastbook import *
from fastai.vision.all import *
from sklearn.metrics import confusion_matrix,accuracy_score
import pandas as pd

In [79]:
path = Path('/content/gdrive/MyDrive/Colab Notebooks/digits/')
csv_path = Path('/content/gdrive/MyDrive/Colab Notebooks/lab4-results.csv')

In [7]:
import_path = Path('/content/gdrive/MyDrive/Colab Notebooks/models/finger_count.pkl')
learn = load_learner(import_path)

In [73]:
pred1 = learn.predict(path/'test/five/00000005.jpg')
pred1

('five',
 TensorBase(0),
 TensorBase([8.0439e-01, 4.3424e-03, 1.7574e-01, 1.5282e-02, 2.4908e-04]))

In [82]:
df = pd.read_csv(csv_path)
df

actual  predicted
0        1          1
1        1          1
2        1          1
3        1          1
4        1          1
5        2          1
6        2          1
7        2          1
8        2          1
9        2          1
10       3          1
11       3          1
12       3          1
13       3          1
14       3          1
15       4          1
16       4          1
17       4          1
18       4          1
19       4          1
20       5          1
21       5          5
22       5          1
23       5          1
24       5          5

In [84]:
confusion_matrix(df.actual, df.predicted)

array([[5, 0, 0, 0, 0],
       [5, 0, 0, 0, 0],
       [5, 0, 0, 0, 0],
       [5, 0, 0, 0, 0],
       [3, 0, 0, 0, 2]])

In [86]:
accuracy_score(df.actual, df.predicted)

0.28

### Question 2: How many predictions did you analyse and how did you select the samples?

I predicted 25 samples which included 5 of each digit. I selected the samples by using the gui created in lab 2 to save 5 samples of each digit, then uploaded the images to Google Drive.

# 2. Summary and Conclusion
Summarize:
- Results from production analysis
- Re-state training, validation results from Lab3

My results coming from the prediction were not very successful. My accuracy was a score of 0.28. The prediction was used with a model that was tested and trained in lab 3. The model used 40 samples of each digit that were then split into train and test data. The images in training were warped to lessen the chance of overfitting of the model. The model was then saved and uploaded to Google Drive to predict in this lab.


Conclusion:
- Does validation accuracy translate to the production environment?
- Are you satisfied with model performance? Why, why not?
- Provide directions for future work, e.g. how could your model and task be improved?

The validation accuracy translates to the production environment. During the validation, the model achieved an accuracy score of about 20-30 percent. This is a similar score that was achieved during prediction in the production environment. I am not very satisfied with the model performance, as I achieved quite a low accuracy score, but it is possible to retrain the model to ahcieve a higher accuracy. For future work, I can retrain the model using methods used in lab 3. From there, I can find a better learning rate than achieves a higher accuracy score, and save that model for use in the production environment. I can also use more training data with different variations so the model has less overfitting to the only data it has seen.

# 3. Reflection
Include a sentence or two about
- what you liked or disliked,
- found interesting, confusing, challangeing, motivating

while working on this assignment.

I liked that we were able to actually see the model that we've been creating the past couple weeks in actual production, and we can see for ourselves how well the model works and if we can improve it. What was motiating was when I saw that my model had a low accuracy score, I hope to improve the model in the future so it works a lot better.
